control-plane/api/v1alpha1/serviceresolver_types.go

```diff
@@ -91,6 +91,9 @@ type ServiceResolverRedirect struct {
 	// Datacenter is the datacenter to resolve the service from instead of the
 	// current one.
 	Datacenter string `json:"datacenter,omitempty"`
+	// Peer is the name of the cluster peer to resolve the service from instead
+	// of the current one.
+	Peer string `json:",omitempty"`
```

Comment:
I think this will require re-generating the yaml.
```suggestion
	Peer string `json:"peer,omitempty"`
```